crates/matrix-sdk-crypto/src/olm/account.rs

```diff
@@ -1111,121 +1106,125 @@ impl Account {
         Ok(())
     }
 
-    /// Try to decrypt an Olm message.
-    ///
-    /// This try to decrypt an Olm message using all the sessions we share
-    /// with the given sender.
-    async fn decrypt_with_existing_sessions(
+    /// Try to decrypt an olm message, creating a new session if necessary.
+    async fn decrypt_olm_message(
+        &mut self,
         store: &Store,
+        sender: &UserId,
         sender_key: Curve25519PublicKey,
         message: &OlmMessage,
-    ) -> OlmResult<Option<(Session, String)>> {
-        let s = store.get_sessions(&sender_key.to_base64()).await?;
+    ) -> Result<(SessionType, String), OlmError> {
+        let existing_sessions = store.get_sessions(&sender_key.to_base64()).await?;
+
+        match message {
+            OlmMessage::Normal(_) => {
+                let session_ids = if let Some(ref sessions) = existing_sessions {
```

Comment:
There's still a `ref` here.

```suggestion
                let session_ids = if let Some(sessions) = existing_sessions {
```